onnxruntime/core/providers/xnnpack/detail/utils.h

```diff
@@ -56,6 +57,50 @@ enum class QuantizedOpType : uint8_t {
   Unknown,
 };
 
+class InternalDataInitializer {
+ public:
+  InternalDataInitializer(const ONNX_NAMESPACE::TensorProto& tensor_proto);
+
+  bool HasExternalData() {
+    return has_external_data_;
+  }
+  const TensorShape& Shape() {
+    return shape_;
+  }
+
+  const DataTypeImpl* Type() {
+    return type_;
+  }
+
+  const void* raw_data() {
+    return unpacked_tensor_.data();
+  }
```

Comment:
Same question for this - when is the original source of the data relevant to a user of the class?

Reply:
emm, DataAsSpan should be more robust than raw_data. removed.